rfcs/2021-05-17-7469-load-balancing-aggregators.md

```diff
@@ -0,0 +1,99 @@
+# RFC 7469 - 2021-05-16 - Scaling and load balancing for Vector aggregators
+
+This RFC describes the need for a tooling and environment agnostic load balancing solution to be bundled with Vector aggregator deployments.
+
+* [Scope](#scope)
+* [Motivation](#motivation)
+* [Internal Proposal](#internal-proposal)
+* [Rationale](#rationale)
+* [Prior Art](#prior-art)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+  * [Do Nothing](#do-nothing)
+  * [Only Client-side Load Balancing](#only-client-side-load-balancing)
+  * [Require a Service Mesh](#require-a-service-mesh)
+* [Outstanding Questions](#outstanding-questions)
+* [Plan Of Attack](#plan-of-attack)
+
+## Scope
+
+Load balancing will be a concern for any `sink` or `source` supported by Vector; some of which can use a general solution (ex: load balancing for our HTTP based `sinks`) and some of which are specific to the component (ex: Kafka or Elasticsearch `sinks`). Due to the breadth of the topic, this RFC will focus on three specific cases for load balancing while also giving consideration to future adoption for other components.
+
+* Vector agent to Vector aggregator
+* Datadog agent to Vector aggregator
+* Syslog agent to Vector aggregator
```

Comment:
Should we mention Vector as a Kafka consumer as out of scope for this RFC? That is a very popular use case, but I think we should separate it since it has nothing to do with reverse proxies, etc.

Reply:
I can mention it, I think we already can scale horizontally for kafka just by the nature of the it. Also on second thought, the ES sink isn't specific at all, so removing that.

Reply:
Is it worth mentioning Kafka more since it's already implemented due to librdkafka and kafka itself?

Reply:
It's popular enough that it's worth calling out, but we do not need to solve for it in this RFC. We can approach Kafka scaling separately.